File: src/main/java/com/trading/modules/order/service/OrderService.java
package com.trading.modules.order.service;

import java.util.List;
import java.util.UUID;
import com.trading.modules.order.dto.CreateOrderRequest;
import com.trading.modules.order.dto.OrderResponse;
import com.trading.modules.order.model.OrderStatus;

public interface OrderService {
    OrderResponse createOrder(CreateOrderRequest request);

    List<OrderResponse> getOrdersByUserId(UUID userId);

    void updateOrderStatus(UUID orderId, OrderStatus status);
}

File: frontend/src/api/orderService.ts
import type { CreateOrderRequest, OrderResponse } from "@/types/order";
import { api } from "./client";

export const orderService = {
    createOrder: async (request: CreateOrderRequest): Promise<OrderResponse> => {
        const { data } = await api.post<OrderResponse>("/orders", request);
        return data;
    },

    getOrdersByUser: async (): Promise<OrderResponse[]> => {
        const { data } = await api.get<OrderResponse[]>("/orders/me");
        return data;
    },
};
